Tasks/VSMobileCenterUpload/vsmobilecenterupload.ts

```diff
@@ -0,0 +1,348 @@
+        responseHandler(defer, err, res, body, () => {
+            tl.debug('-- Symbol uploaded.');
+            defer.resolve();
+        });
+    }));
+
+    return defer.promise;
+}
+
+function commitSymbols(apiServer: string, apiVersion: string, appSlug: string, symbol_upload_id: string, token: string): Q.Promise<void> {
+    tl.debug("-- Finishing uploading symbols...");
+    let defer = Q.defer<void>();
+    let commitSymbolsUrl: string = `${apiServer}/${apiVersion}/apps/${appSlug}/symbol_uploads/${symbol_upload_id}`;
+    tl.debug(`---- url: ${commitSymbolsUrl}`);
+    let headers = {
+        "X-API-Token": token
+    };
+
+    let commitBody = { "status": "committed" };
+
+    request.patch({ url: commitSymbolsUrl, headers: headers, json: commitBody }, (err, res, body) => {
+        responseHandler(defer, err, res, body, () => {
+            defer.resolve();
+        });
+    })
+
+    return defer.promise;
+}
+
+async function run() {
+    try {
+        tl.setResourcePath(path.join( __dirname, 'task.json'));
+
+        // Get build inputs
+        let apiEndpointData = getEndpointDetails('serverEndpoint');
+        let apiToken: string = apiEndpointData.authToken;
+        let apiServer: string = apiEndpointData.apiServer;
+        let apiVersion: string = apiEndpointData.apiVersion;
+    
+        var effectiveApiServer = process.env['SONOMA_API_SERVER'] || apiServer;
+        var effectiveApiVersion = process.env['SONOMA_API_VERSION'] || apiVersion;
+        
+        tl.debug(`Effective API Url: ${effectiveApiServer}/${effectiveApiVersion}`);
+        
+        let appSlug: string = tl.getInput('appSlug', true);
+        let appFilePattern: string = tl.getInput('app', true);
+        let symbolsType: string = tl.getInput('symbolsType', false);
+        let symbolVariableName = null;
+        switch (symbolsType) {
+            case "Apple": 
+                symbolVariableName = "dsymPath";
+                break;
+            case "AndroidJava":
+                symbolVariableName = "mappingTxtPath";
+                break;
+            default: 
+                symbolVariableName = "symbolsPath";
+        }
+        let symbolsPathPattern: string = tl.getInput(symbolVariableName, false);
+        let packParentFolder: boolean = tl.getBoolInput('packParentFolder', false);
+
+        let releaseNotesSelection = tl.getInput('releaseNotesSelection', true);
+        let releaseNotes: string = null;
+        if (releaseNotesSelection === 'file') {
+            let releaseNotesFile = tl.getPathInput('releaseNotesFile', true, true);
+            releaseNotes = fs.readFileSync(releaseNotesFile).toString('utf8');
+        } else {
+            releaseNotes = tl.getInput('releaseNotesInput', true);
+        }
+
+        let distributionGroupId = tl.getInput('distributionGroupId', false) || '00000000-0000-0000-0000-000000000000';
+        tl.debug(`Effective distribution_group_id: ${distributionGroupId}`);
```

Comment:
I opted not to validate this as this is a contract between the user and mobile center API.  I don't want to get in between and do my own validation just in case the API evolves in the future.